temporian/core/operators/tick_calendar.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2021 Google LLC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""TickCalendar operator class and public API function definitions."""
+from typing import Union
+
+from temporian.core import operator_lib
+from temporian.core.compilation import compile
+from temporian.core.data.node import (
+    EventSetNode,
+    create_node_new_features_new_sampling,
+)
+from temporian.core.operators.base import Operator
+from temporian.core.typing import EventSetOrNode
+from temporian.proto import core_pb2 as pb
+from temporian.utils.typecheck import typecheck
+
+
+class TickCalendar(Operator):
+    def __init__(
+        self,
+        input: EventSetNode,
+        min_second: int,
+        min_minute: int,
+        min_hour: int,
+        min_mday: int,
+        min_month: int,
+        min_wday: int,
+        max_second: int,
+        max_minute: int,
+        max_hour: int,
+        max_mday: int,
+        max_month: int,
+        max_wday: int,
+    ):
+        super().__init__()
+        if not input.schema.is_unix_timestamp:
+            raise ValueError(
+                "Can only use tick_calendar on unix timestamp samplings"
+            )
+
+        # Attributes
+        self._min_second = min_second
+        self._max_second = max_second
```

Comment:
This min/max logic is kind of a (great) implementation detail.
However, it means that we cannot later change / improve / extend the implementation (since OP will possibly be serialized).

If the conversion  from user-input to min/max would be done in the constructor of TickCalendar, instead of in tick_calendar, the code would be more extensive.

Reply:
The serialization code would be more complex if we want to allow both `'*'` and `int` for the attribute types.
We'd need to have an optional flag (e.g. `is_wildcard`) plus an optional integer. Or use `-1` as a special value which may collide with actual values in the future (e.g: if we allow `mday=-1`).
Also, if we want to support ranges in the future, we'd also need to include more optional attributes.

So, I think that saving the range is actually simpler and more generic than  what is currently supported by the api. We can always add optional attributes later for other use cases, right?
Let's discuss it tomorrow.